Firebase/Messaging/FIRMessaging.m

```diff
@@ -756,13 +756,19 @@ - (void)subscribeToTopic:(NSString *)topic
                            @"subscribeToTopic.",
                            topic, [FIRMessagingPubSub removePrefixFromTopic:topic]);
   }
-  if (!self.defaultFcmToken.length) {
-    FIRMessagingLoggerWarn(kFIRMessagingMessageCodeMessaging010,
-                           @"The subscription operation is suspended because you don't have a "
-                           @"token. The operation will resume once you get an FCM token.");
-  }
-  NSString *normalizeTopic = [[self class] normalizeTopic:topic];
-  if (normalizeTopic.length) {
+  [self.instanceID instanceIDWithHandler:^(FIRInstanceIDResult *_Nullable result,
+                                           NSError *_Nullable error) {
+    if (!result.token || error) {
+      FIRMessagingLoggerError(
+          kFIRMessagingMessageCodeMessaging010,
+          @"The subscription operation failed due to an error getting the FCM token: %@.", error);
+      if (completion) {
+        completion(error);
+      }
+      return;
+    }
+    NSString *normalizeTopic = [[self class] normalizeTopic:topic];
+    if (normalizeTopic.length) {
     [self.pubsub subscribeToTopic:normalizeTopic handler:completion];
```

Comment:
Nit: When there is no an explicit API convention on the block usage, I usually try to capture a weak self reference and declare a local strong reference in the block just in case. With current implementation of `instanceIDWithHandler`capturing `self` should be fine, but if it changes, then this code will be still safe to use.